cpp/src/arrow/compute/exec.h

```diff
@@ -30,7 +30,7 @@
 
 #include "arrow/array/data.h"
 #include "arrow/compute/exec/expression.h"
-#include "arrow/compute/type_fwd.h"
+#include "arrow/compute/registry.h"
```

Comment:
I don't think this change is necessary actually?

Reply:
Yes, it becomes unnecessary when the forward-declaration are moved to `compute/type_fwd.h`, as you suggested below.